pennylane/transforms/convert_to_numpy_parameters.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains preprocessings steps that may be called internally
+during execution.
+"""
+import copy
+
+import pennylane as qml
+from pennylane import math
+from pennylane.tape import QuantumScript
+
+
+def _convert_op_to_numpy_data(op: qml.operation.Operator) -> qml.operation.Operator:
+    if math.get_interface(*op.data) == "numpy":
+        return op
+    # Use operator method to change parameters when it become available
+    copied_op = copy.copy(op)
+    copied_op.data = math.unwrap(op.data)
+    return copied_op
+
+
+def _convert_measurement_to_numpy_data(
+    m: qml.measurements.MeasurementProcess,
+) -> qml.measurements.MeasurementProcess:
+    if m.obs is None or math.get_interface(*m.obs.data) == "numpy":
+        return m
+    # Use measurement method to change parameters when it becomes available
+    copied_m = copy.copy(m)
+    copied_m.obs.data = math.unwrap(m.obs.data)
+    return copied_m
+
+
+# pylint: disable=protected-access
+def convert_to_numpy_parameters(circuit: QuantumScript) -> QuantumScript:
+    """Transforms a circuit to one with purely numpy parameters.
+
+    Args:
+        circuit (QuantumScript): a circuit with parameters of any interface
+
+    Returns:
+        QuantumScript: A circuit with purely numpy parameters
+
+    See also :class:`pennylane.tape.Unwrap`.  ``convert_to_numpy_parameters`` function creates a new :class:`pennylane.tape.QuantumScript`
```

Comment:
can we use the Sphinx seealso directive here? The formatting in the docs look a little busy here imo